src/node_zlib.cc

```diff
@@ -974,6 +982,15 @@ CompressionError ZlibContext::Init(
     window_bits_ *= -1;
   }
 
+  dictionary_ = std::move(dictionary);
+}
+
+bool ZlibContext::InitZlib() {
+  if (zlib_init_done_) {
+    return false;
+  }
+  zlib_init_done_ = true;
```

Comment:
This variable is written here, which as I understand it happens in the thread pool, but is read in `Close()`, which happens on the main thread.

Is that a race condition?

Reply:
Fair point. Let me think how to avoid it.

Reply:
I have protected the field with a mutex. See b98c0143353c2cfddbb941be5258879f74d27faa

Although, I'm not sure if that's a good approach, as it means that `Close()` (and thus the event loop) may block for the duration of `InitZlib()`. On the other hand, the original code implied that contents of `InitZlib()` were executed on the event loop thread (yet, without the overhead of the mutex). WDYT?